compiler/qsc_hir/src/ty.rs

```diff
@@ -479,6 +479,28 @@ impl Udt {
         Some(field)
     }
 
+    /// The field with the given name. Returns [None] if this user-defined type does not
+    /// have a field with the given name.
+    #[must_use]
+    pub fn find_field_by_name(&self, name: &str) -> Option<&UdtField> {
+        Self::find_field_by_name_rec(&self.definition, name)
```

Comment:
nitpick: if the helper function doesn't use `self` then it can be put outside of the `impl` block and just be a local function for the file. That would remove the need to write `Self::` when calling it.

Reply:
I considered that while working on this, but this is such a large file with so many different structs in it, it would be easy for a loose function to get lost. It makes more sense to me here that the functions be inside the class and right next to their related functions.